components/brave_news/rust/lib.rs

```diff
@@ -26,10 +27,70 @@ mod ffi {
     }
 
     extern "Rust" {
+        fn strip_html(subject: &str) -> String;
         fn parse_feed_bytes(source: &[u8], output: &mut FeedData) -> bool;
     }
 }
 
+// Note: This function isn't likely to be perfect, but it does guarantee that
+// there will be no HTML tags in the output (as it strips both '<' and '>' from
+// from the output).
+fn strip_html(subject: &str) -> String {
+    let mut output = String::with_capacity(subject.len());
+    let mut depth_tag = 0;
+    let mut depth_comment = 0;
+
+    // We want to know the last 3 characters so we can see if they were --> for
+    // closing a comment.
+    let last_chars_size = 3;
+    let mut last_chars = String::with_capacity(last_chars_size);
+    let mut iterator = subject.chars();
+
+    while let Some(c) = iterator.next() {
```

Comment:
How do we expect this to respond to HTML entities such as `&lt;`? 

For example, I'm seeing this test passes, but I'm not sure if this is the expected behavior we want for this case:

`EXPECT_EQ("&lt;pThis is paragraph an HTML entity.", Strip("&lt;p>This is paragraph an HTML entity.</p>"));`

Reply:
I think this is the correct behaviour. `&lt;` is html-escaped, so it represents part of the normal text in the document fragment. We want to strip out active tags. This would only be a problem if the code were resolving entities like this *after* stripping tags.

Reply:
:+1: to this, but let's add that test case in